source/simulator/core.cc

```diff
@@ -75,6 +76,17 @@ namespace aspect
 
       return false;
     }
+
+    template <int dim>
+    std::vector<dealii::Variable<dim> > get_variables(const Parameters<dim> &parameters,
+                                                      SimulatorSignals<dim> &signals)
```

Comment:
Am I getting this right that this function creates the list of variables, and not just returns them?
In this case, should we call it something like `create_list_of_variables` or `set_up_variables`?